changelog.d/20241101_153924_chris_function_name_code.rst

```diff
@@ -0,0 +1,5 @@
+Deprecated
+^^^^^^^^^^
+
+- The ``function_name`` argument to ``Client.register_function`` is not in use, and has
+  now been deprecated. Function names are determined by the function's ``__name__``.
```

Comment:
"is not in use" ... have we been using it to date?  Or has it been a vestigial argument?

Reply:
Vestigial - I've clarified the changelog message to explain that